FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.DataClient/Api/ICredentialProvider.cs

```diff
@@ -0,0 +1,14 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using Azure.Core;
+
+namespace Microsoft.Health.Fhir.Synapse.DataClient.Api
+{
+    public interface ICredentialProvider
```

Comment:
We want to enable two types of credentials, `ExternalTokenCredentialProvider` and `InternalTokenCredentialProvider` (both implemented by DefaultTokenCredential in OSS), for current FHIR and Data lake storage, they both utilize the `ExternalTokenCredentialProvider`